engine/gameobject/src/gameobject/test/wscript

```diff
@@ -27,8 +28,16 @@ create_simple_protoc_task('input_target', '.it_pb', '.it', 'TestGameObjectDDF.In
 create_simple_protoc_task('reload_target', '.rt_pb', '.rt', 'TestGameObjectDDF.ReloadTarget', before='c cxx', proto_file = '../src/gameobject/test/reload/test_gameobject_reload_ddf.proto')
 create_simple_protoc_task('no_user_data_res', '.no_user_data', '.no_user_datac', 'TestGameObjectDDF.NoUserDataResource', before='c cxx', proto_file = '../src/gameobject/test/props/test_gameobject_props_ddf.proto')
 
+
+def clear_caches():
+    importlib_lock.acquire()
+    try:
+        importlib.invalidate_caches() # make sure any modules created after the processes started will be found
+    finally:
+        importlib_lock.release()
```

Comment:
When I tried to print out the cached modules, I got an error that the dictionary I was iterating was being modified.
Which in turn suggests that it might be a multi threading issue (on windows only?), which could also explain the intermittency of the problem.

In any case, this code isn't "worse" than before, and we'll see if it helps.